### DISCUSSION TITLE
Python client connects to URL other than the one specified

### DISCUSSION THREAD

**Discussion by rgov** [miguelgrinberg/python-socketio#819] [Q&A]
I am running python-socketio version 5.3.0.

```python
await sio.connect('https://localhost/my-app/', namespaces=['/ns'])
```

This inexplicably tries to load the URL `https://localhost/socket.io/?transport=polling&EIO=4`, ignoring the path `/my-app/` that I provided in the `url` parameter.

This works:

```python
await sio.connect('https://localhost', socketio_path='/my-app/')
```

The `socketio_path` parameter is not documented on https://python-socketio.readthedocs.io/en/latest/client.html#connecting-to-a-server

**Comment by miguelgrinberg** [comment]
> This inexplicably tries to load 

It may seem inexplicable, but that is how the Socket.IO protocol has always worked.
 
> The socketio_path parameter is not documented

The `socketio_path` argument is documented in the reference section: https://python-socketio.readthedocs.io/en/latest/api.html#socketio.Client.connect. Note that this option is very rarely used, the vast majority of Socket.IO servers use the default Socket.IO endpoint, so the client's default does not need to be changed.
